.github/workflows/doc.yml

```diff
@@ -16,7 +16,6 @@ jobs:
       - uses: actions/checkout@v4
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
-      - name: Install cargo make
-        run: cargo install cargo-make
+      - uses: davidB/rust-cargo-make@v1
```

Comment:
I'm still not sure we should use `davidB/rust-cargo-make`. It doesn't seem to be a very popular package (e.g., just 31 stars on Github) and I'm always weary of using such packages, mostly due to the risk of supply chain attacks.

How much faster does the CI run because of this?

Reply:
It reduced the CI running time by 1 min 39 sec approximately: 
<img width="1321" alt="SCR-20240419-odbd" src="https://github.com/0xPolygonMiden/miden-node/assets/42912740/b3861760-a7b3-4fee-93b0-7b6cd8217206">

From 1 min40 sec to 1 sec thanks to a simple use of gh actions. 

But I believe that 1 min 40 sec gain in the CI is a marginal earning compared to the risk of a supply chain attack compromising the security of our chain. 

What is your point of view? Should we revert?

Reply:
1 min 40 sec is not negligible - but also, not something that I think will impact our processes much.

On the other hand, the potential security risk is rather small (it only affects the CI) - so, maybe not worth worrying about it too much.

Overall, I'm on the fence - but leaning toward keeping this as they are now (i.e., keep using `davidB/rust-cargo-make`).